core/src/gl/ogl_render_device.cpp

```diff
@@ -2009,12 +2009,14 @@ Texture2DArray OGLRenderDevice::createTexture2DArray(const Texture2DArrayDesc& d
     GLuint id;
     glGenTextures(1, &id);
     glBindTexture(GL_TEXTURE_2D_ARRAY, id);
-    glTexStorage3D(GL_TEXTURE_2D_ARRAY, static_cast<GLsizei>(desc.mipLevelCount), internalFormat,
-                   static_cast<GLsizei>(desc.width), static_cast<GLsizei>(desc.height),
-                   static_cast<GLsizei>(desc.size));
+    for (std::size_t j = 0; j < desc.mipLevelCount; ++j)
+    {
+        glTexImage3D(GL_TEXTURE_2D_ARRAY, static_cast<GLint>(j), internalFormat, static_cast<GLsizei>(desc.width),
```

Comment:
```suggestion
        glTexImage3D(GL_TEXTURE_2D_ARRAY, static_cast<GLint>(j), static_cast<GLint>(internalFormat), static_cast<GLsizei>(desc.width),
```
On clang is not compiling without this static_cast